src/api/admin.rs

```diff
@@ -31,8 +31,12 @@ use crate::{
     CONFIG, VERSION,
 };
 
+#[allow(clippy::nonminimal_bool)]
 pub fn routes() -> Vec<Route> {
-    if !CONFIG.disable_admin_token() && !CONFIG.is_admin_token_set() {
+    if !CONFIG.disable_admin_token()
+        && !CONFIG.is_admin_token_set()
+        && !(CONFIG.sso_enabled() && CONFIG.sso_roles_enabled())
```

Comment:
This could be simplified to this:

```suggestion
if !(disable_admin_token || is_admin_token_set || sso_enabled && sso_roles_enabled)
```